### PR TITLE
Mapping files for Hercules DJControl MIX

### PR DESCRIPTION
**Pull request by DJPhatso**
Based on Hercules DJControl Starlight mapping released with Mixxx v2.3.0:
 -Remapped LOOP and SAMPLER section according to DJControl MIX layout; 
-Added Master Volume and Headphone Volume; 
-Removed superfluous LED configuration (not present on DJControl MIX).

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
The Pre-Commit check fails. You can fix this by using pre-commit.patch artifact from the CI build:
git apply pre-commit.patch
or you install pre-commit locally on your system: https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking

**Comment by DJPhatso** [comment]
>  You can fix this by using pre-commit.patch artifact from the CI build: git apply pre-commit.patch 

Not quite sure how to go about doing that.
Tried installing pre-commit locally, but it looks to be difficult with the workplace restrictions.

**Comment by JoergAtGithub** [comment]
Than download this file:
grafik
and execute
`git apply pre-commit.patch`

**Comment by DJPhatso** [comment]
Done.

**Comment by JoergAtGithub** [comment]
Please try again, this wasn't successful

**Comment by DJPhatso** [comment]
I re-applied the patch. Now trying to resync to commit again, but that doesn't seem to work.

**Comment by JoergAtGithub** [comment]
Please set up automaic code checking: https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking

**Comment by DJPhatso** [comment]
As mentioned before, I tried these steps before, but I'm doing this on my work computer, and the admin restrictions are not allowing me to install everything properly. Planning on eventually starting anew on another personal pc, but this is not possible currently.

Is there any work around ? This seems complicated just to submit a mapping.

**Comment by JoergAtGithub** [comment]
You don't need to install anything except git. Than download the `pre-commit.patch` from the failing GitHub build and call `git apply pre-commit.patch`

**Comment by DJPhatso** [comment]
If that would have worked, we wouldn't be here...

I'm in the process of installing and configuring everything as best as I can on a Linux system, and use that only for submission, but that in itself is something I could do without ATM.

**Comment by Swiftb0y** [comment]
Hey there @DJPhatso. Have you seen my PR https://github.com/DJPhatso/mixxx/pull/7 that fixes the pre-commit issues? If you merge that, we can start looking a more closer look at this PR. what do you think?

**Comment by DJPhatso** [comment]
Just checking if this is in-line to be included in the next release or if there is anything left to do.

**Comment by JoergAtGithub** [comment]
We need also the documentation page for the manual, see: https://github.com/mixxxdj/mixxx/wiki/Contributing-Mappings#user-content-documenting-the-mapping

File: res/controllers/Hercules-DJControl-MIX-scripts.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", 0.6);
+        engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", 0.6);
+        engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", 0.6);
+        engine.setParameter("[EffectRack1_EffectUnit1]", "mix", 1);
+        engine.setParameter("[EffectRack1_EffectUnit2]", "mix", 1);
+
+        // Ask the controller to send all current knob/slider values over MIDI, which will update
+        // the corresponding GUI controls in MIXXX.
+        midi.sendShortMsg(0xB0, 0x7F, 0x7F);
+    };
```
Comment (JoergAtGithub) [inline res/controllers/Hercules-DJControl-MIX-scripts.js:59]:
```suggestion
    }
```
Please delete this semicolon

File: res/controllers/Hercules-DJControl-MIX-scripts.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        // ****************************************************************************
+        // * Mixxx mapping script file for the Hercules DJControl MIX.
+        // * Author: DJ Phatso and Kerrick Staley
+        // * Version 1 (Jan 2023)
+        // * Based on Hercules DJControl Starlight mapping released with Mixxx v2.3.0
+        // *  -Remapped LOOP and SAMPLER section according to DJControl MIX layout
+        // *  -Added Master Volume and Headphone Volume
+        // *  -Removed superfluous LED configuration (not present on DJControl MIX)
+        // * Forum: https://mixxx.discourse.group/t/hercules-contrl-mix-mapping/26581/
+        // * Wiki: https://mixxx.org/wiki/doku.php/
```
Comment (JoergAtGithub) [inline res/controllers/Hercules-DJControl-MIX-scripts.js:27]:
```suggestion
```
Please delete this dead link

**Review by JoergAtGithub — CHANGES_REQUESTED** [review]
Overall it's looking good, but I listed some minor findings, which should be corrected before merging this.

**Review by JoergAtGithub — APPROVED** [review]

**Comment by JoergAtGithub** [comment]
LGTM! Thank you!

**Comment by daschuer** [comment]
Hi @DJPhatso, thank you for your contribution.
It looks like we have missed to ask for your permission to distribute your code before merge. Can you catch that up, sign https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ and comment here when done? I will then put you the Mixxx 2.4 contributor list in the about box. May I use your real name?

**Comment by DJPhatso** [comment]
Filled the form, and yes, you use my name.

**Pull request merged**